### PR TITLE
Remove Apple OpenCL tests from CI

### PR DESCRIPTION
**Pull request by majosm**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -83,6 +83,9 @@ jobs:
             MINIFORGE_INSTALL_DIR=.miniforge3
             . "$MINIFORGE_INSTALL_DIR/bin/activate" testing
             cd test
+            if [[ ${{ matrix.os }} == "macos-latest" ]]; then
+              export PYOPENCL_TEST=0:0
```
Comment (inducer) [inline .github/workflows/ci.yaml:87]:
```suggestion
              export PYOPENCL_TEST=port:pthread
```

**Comment by majosm** [comment]
Well this is a little strange. It's only running the pocl tests now, but it's still taking just as long?

This branch:
```
================= 74 passed, 378 warnings in 892.16s (0:14:52) =================
```

Last night's CI:
```
================ 148 passed, 756 warnings in 853.80s (0:14:13) =================
```

**Comment by majosm** [comment]
Let's find out (re-running both).

**Comment by majosm** [comment]
There we go.

This branch:
```
================= 74 passed, 378 warnings in 580.88s (0:09:40) =================
```

master:
```
================ 148 passed, 756 warnings in 992.80s (0:16:32) =================
```

**Comment by majosm** [comment]
Just occurred to me that my other post was ambiguous: "Last night's CI" should be read as "Last night's CI (on master)".

**Review by inducer — APPROVED** [review]

**Pull request merged**